Dockerfile

```diff
@@ -1,6 +1,6 @@
-# syntax = docker/dockerfile:1.4
+# syntax=docker/dockerfile:1
 
-ARG GO_VERSION
+ARG GO_VERSION=1.23.2
```

Comment:
I think it's better and easier to maintain without a default. Relying on [the bash script to build](https://github.com/artefactual-sdps/preprocessing-sfa/blob/main/hack/build_docker.sh) and the .go-version. Or, like the recent changes in Enduro, [reading from go.mod directly](https://github.com/artefactual-sdps/enduro/blob/main/hack/build_docker.sh#L45-L49).

Reply:
Not supplying a default value for a required ARG value goes against best practice: https://docs.docker.com/reference/build-checks/invalid-default-arg-in-from/ and the Docker linter I'm using complains about it.

Reply:
Adding a default value also doesn't change the results when building the Docker image from the bash script or in the Enduro case.

Reply:
I know, it's just sad to have to update the default, but I can see their point.

Reply:
Yeah, it is unfortunate that it's one more place we have to update the Go version. :(